js/main.js

```diff
@@ -1,45 +1,65 @@
-let books = [];
-const booksSection = document.getElementById('books');
-const form = document.getElementById('book_form');
+class AwesomeBooks {
+  books = [];
+
+  saveBooks() {
+    const booksList = JSON.stringify(this.books);
+    localStorage.setItem('books', booksList);
+  }
+
+  addBook(title, author) {
+    const book = {
+      title,
+      author,
+      id: Date.now(),
+    };
+    this.books.unshift(book);
+    this.saveBooks();
+  }
```

Comment:
- [ **OPTIONAL**] This method works well to add a book to the books array but it mutates the original array. The `unshift` does not return a new copy of the `books` array. Mutation may lead to unexpected and hard-to-debug issues. Please let's always consider using methods that do not mutate the original array like concat(), filter(), slice().